test/scripts/feature-flags.js

```diff
@@ -0,0 +1,258 @@
+    });
+
+    it(`${command} fails if tokenCodeClients contains non-boolean enableTestEmails`, () => {
+      return cp.execAsync(
+        `echo '{"tokenCodeClients":{"0123456789abcdef":{"enableTestEmails":1,"groups":["treatment"],"name":"wibble","rolloutRate":1}}}' | node scripts/feature-flags ${command}`,
+        { cwd }
+      )
+        .then(() => assert(false, 'script should have failed'), () => {});
+    });
+
+    it(`${command} fails if tokenCodeClients contains empty groups string`, () => {
+      return cp.execAsync(
+        `echo '{"tokenCodeClients":{"0123456789abcdef":{"enableTestEmails":true,"groups":[""],"name":"wibble","rolloutRate":1}}}' | node scripts/feature-flags ${command}`,
+        { cwd }
+      )
+        .then(() => assert(false, 'script should have failed'), () => {});
+    });
+
+    it(`${command} fails if tokenCodeClients contains empty name`, () => {
+      return cp.execAsync(
+        `echo '{"tokenCodeClients":{"0123456789abcdef":{"enableTestEmails":true,"groups":["treatment"],"name":"","rolloutRate":1}}}' | node scripts/feature-flags ${command}`,
+        { cwd }
+      )
+        .then(() => assert(false, 'script should have failed'), () => {});
+    });
+
+    it(`${command} fails if tokenCodeClients contains rolloutRate greater than 1`, () => {
+      return cp.execAsync(
+        `echo '{"tokenCodeClients":{"0123456789abcdef":{"enableTestEmails":true,"groups":["treatment"],"name":"wibble","rolloutRate":1.1}}}' | node scripts/feature-flags ${command}`,
+        { cwd }
+      )
+        .then(() => assert(false, 'script should have failed'), () => {});
+    });
+  });
+
+  it('clear does not fail', () => {
+    return cp.execAsync('node scripts/feature-flags revert', { cwd });
+  });
+
+  it('revert does not fail', () => {
+    return cp.execAsync('node scripts/feature-flags revert', { cwd });
+  });
+
+  describe('write:', () => {
+    let flags;
+
+    beforeEach(() => {
+      flags = {
+        communicationPrefLanguages: [ "en", "fr" ],
+        metricsSampleRate: 1,
+      };
+      return cp.execAsync(`echo '${JSON.stringify(flags)}' | node scripts/feature-flags write`, { cwd });
+    });
+
+    it('read prints the flags to stdout', async () => {
+      const stdout = await cp.execAsync('node scripts/feature-flags read', { cwd });
+      assert.equal(stdout, `${JSON.stringify(flags, null, '  ')}\n`);
+    });
+
+    describe('merge:', () => {
+      let moreFlags;
+
+      beforeEach(() => {
+        moreFlags = {
+          communicationPrefLanguages: [ "en", "de" ],
+          sentrySampleRate: 0,
+        };
+        return cp.execAsync(`echo '${JSON.stringify(moreFlags)}' | node scripts/feature-flags merge`, { cwd });
+      });
+
+      it('read prints the merged flags to stdout', async () => {
+        const stdout = await cp.execAsync('node scripts/feature-flags read', { cwd });
+        assert.equal(stdout, `${JSON.stringify({ ...flags, ...moreFlags }, null, '  ')}\n`);
+      });
```

Comment:
Should there be a test to assert correct behavior if any of the new input is invalid?

Reply:
Not sure I understand this question. Is it not covered by the failure assertions above?